src/layouts/Settings/Settings.tsx

```diff
@@ -98,14 +98,35 @@ const SettingsForm = ({ settings, isError }: SettingsFormProps) => {
   const { formState, reset, getValues } = methods
   const { isDirty, dirtyFields } = formState
   const successToast = useSuccessToast()
+  const errorToast = useErrorToast()
   const { setIsDirty } = useDirtyFieldContext()
   const {
     mutateAsync: updateSettings,
     error: updateSettingsError,
     isLoading,
     isSuccess,
-  } = useUpdateSettings(siteName)
-  const errorToast = useErrorToast()
+  } = useUpdateSettings(
+    siteName,
+    () => {
+      successToast({
+        id: "update-settings-success",
+        title: "Success",
+        description: "Site settings have been updated",
+      })
+    },
+    () => {
+      const errorMessage =
+        updateSettingsError?.response?.status === 409
+          ? "Your site settings have recently been changed by another user. You can choose to either override their changes, or go back to editing. We recommend you to make a copy of your changes elsewhere, and come back later to reconcile your changes."
+          : `Site settings could not be updated. ${DEFAULT_RETRY_MSG}`
+      return errorToast({
+        id: "update-settings-error",
+        title: "Error",
+        description: errorMessage,
+      })
+    }
+  )
+
```

Comment:
would be good if we shifted all these to the hook and mark as optional but is a good to have bah...

Reply:
oof merge liao :/ + I abit dont get this what do you mean  by shift to hook ah?